packages/components/fields/async-creatable-select-field/src/async-creatable-select-field.tsx

```diff
@@ -390,8 +390,7 @@ export default class AsyncCreatableSelectField extends Component<
 
     if (this.props.hintIcon) {
       warning(
-        typeof this.props.hint === 'string' ||
-          isValidElement(typeof this.props.hint),
+        typeof this.props.hint === 'string' || isValidElement(this.props.hint),
```

Comment:
Makes sense. Thanks.

I presume `isValidElement` comes from [here](https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react/src/jsx/ReactJSXElementValidator.js#L65). There is also a `isValidElementType` defined [here](https://github.com/facebook/react/blob/main/packages/shared/isValidElementType.js#L39). 

I just wanted to verify that the need the `typeof ''` check in [this sandbox](https://codesandbox.io/s/dreamy-edison-2xkz0n?file=/src/App.js) cause `isValidElementType` looked like it covers strings.